libcontainer/cgroups/file.go

```diff
@@ -147,7 +149,7 @@ func openFile(dir, file string, flags int) (*os.File, error) {
 		// TODO: if such usage will ever be common, amend this
 		// to reopen cgroupFd and retry openat2.
 		fdStr := strconv.Itoa(cgroupFd)
-		fdDest, _ := os.Readlink("/proc/self/fd/" + fdStr)
+		fdDest, _ := os.Readlink(utils.ProcThreadSelf("fd/" + fdStr))
```

Comment:
Why doesn't need `lockOSThread` here?

Reply:
The readlink happens atomically, so we don't need to worry about being rescheduled and the original thread being killed.

Reply:
It turns out we do need locking here for pre-3.17 kernels.